pkg/linters/componentinterface/main.go

```diff
@@ -0,0 +1,87 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package componentinterface
+
+import (
+	"go/ast"
+	"strings"
+
+	"github.com/golangci/plugin-module-register/register"
+	"golang.org/x/tools/go/analysis"
+)
+
+func init() {
+	register.Plugin("componentinterface", New)
+}
+
+type componentInterfacePlugin struct {
+}
+
+func New(_ any) (register.LinterPlugin, error) {
+	return &componentInterfacePlugin{}, nil
+}
+
+func (f *componentInterfacePlugin) BuildAnalyzers() ([]*analysis.Analyzer, error) {
+	return []*analysis.Analyzer{
+		{
+			Name: "componentinterface",
+			Doc:  "ensure Component interface do not import third party packages",
+			Run:  f.run,
+		},
+	}, nil
+}
+
+func (f *componentInterfacePlugin) GetLoadMode() string {
+	return register.LoadModeSyntax
+}
+
+func (f *componentInterfacePlugin) run(pass *analysis.Pass) (interface{}, error) {
+	for _, f := range pass.Files {
+		var componentInterface bool
+
+		ast.Inspect(f, func(node ast.Node) bool {
+			switch t := node.(type) {
+			// find variable declarations
+			case *ast.TypeSpec:
+				// which are public
+				if t.Name.IsExported() {
+					switch t.Type.(type) {
+					// and are interfaces
+					case *ast.InterfaceType:
+						if t.Name.Name == "Component" {
+							componentInterface = true
+							return false
+						}
+					}
+				}
+			}
+			return true
+		})
+
+		if componentInterface {
+			for _, imp := range f.Imports {
+				if strings.Contains(imp.Path.Value, ".") && strings.Contains(imp.Path.Value, "/") {
```

Comment:
It would be nice to document why you check that the import contains `.` (to exclude go runtime types?) and `/` (?).